kong/db/dao/workspaces.lua

```diff
@@ -18,4 +26,14 @@ function Workspaces:truncate()
 end
 
 
+function Workspaces:select_by_name(key, options)
+  if kong.configuration.database == "off" and key == "default" then
+    -- it should be a table, not a single string
+    return { id = lmdb.get(DECLARATIVE_DEFAULT_WORKSPACE_KEY) or DECLARATIVE_DEFAULT_WORKSPACE_ID, }
```

Comment:
```suggestion
    return { id = lmdb.get(DECLARATIVE_DEFAULT_WORKSPACE_KEY), }
```


As talked with @dndx @chronolaw , we'd better insert the DECLARATIVE_DEFAULT_WORKSPACE_ID into lmdb if dbless kong instance starts. Then we don't need to return a hardcode value here directly.

Left a comment here to block the PR, need to understand how default workspace(id) works in Kong master.

Reply:
For hybrid mode, the default workspace works as following logic:

1. Before we starts kong(cp/dp), we run `kong migrations bootstrap`, it will create the workspaces table, and insert the default workspace to postgres.
2. When we start kong CP, it loads all entities from postgres, including default workspace.
3. When we start kong DPs, they load all entities from CP, including default workspaces. And these loaded entities from CP will overwrite the default ones from lmdb which is loaded from declarative config.

Reply:
for master, we dont load a lmdb, so we need to return a constant value now. left a TODO, we might load lmdb in master instead of workers in the further.